lib/smart_proxy_ansible/runner/ansible_runner.rb

```diff
@@ -83,10 +83,18 @@ def handle_host_event(hostname, event)
       def handle_broadcast_data(event)
         log_event("broadcast", event)
         if event['event'] == 'playbook_on_stats'
+          failures = event.dig('event_data', 'failures') || {}
           header, *rows = event['stdout'].strip.lines.map(&:chomp)
           @outputs.keys.select { |key| key.is_a? String }.each do |host|
             line = rows.find { |row| row =~ /#{host}/ }
             publish_data_for(host, [header, line].join("\n"), 'stdout')
+
+            # If the task has been rescued, it won't consider a failure
+            if @exit_statuses[host] && @exit_statuses[host] != 0
+              if failures[host].nil? || failures[host] <= 0
+                publish_exit_status_for(host, 0)
+              end
+            end
```

Comment:
Can we makes this a single `if` with a rather large condition instead of two nested ifs?

Reply:
Yes. we can. I just didn't like to have a long line or 2 lines condition

Reply:
What if we rely on `nil.to_i` returning `0` and turn this into `@exit_statuses[host].to_i != 0 && failures[host].to_i <= 0`?

Reply:
Done! Thanks for the suggestion.